ext/filter/filter.c

```diff
@@ -524,14 +524,16 @@ PHP_FUNCTION(filter_has_var)
 }
 /* }}} */
 
-static void php_filter_call(zval *filtered, zend_long filter, zval *filter_args, const int copy, zend_long filter_flags) /* {{{ */
-{
+static void php_filter_call(
+	zval *filtered, zend_long filter, HashTable *filter_args_ht, zend_long filter_args_long,
+	const int copy, zend_long filter_flags
+) /* {{{ */ {
 	zval *options = NULL;
 	zval *option;
 	char *charset = NULL;
 
-	if (filter_args && Z_TYPE_P(filter_args) != IS_ARRAY) {
-		zend_long lval = zval_get_long(filter_args);
+	if (filter_args_long) {
```

Comment:
Shouldn't the check here rather the `!filter_args_ht`?

Reply:
This change would make 3 tests fail:
- in `ext/filter/tests/039.phpt`: `var_dump(filter_var_array(array()));` would become `false` rather than `array()`
- in `ext/filter/tests/057.php`: a bunch of things would change, e.g. `var_dump(filter_input_array(INPUT_POST, null));` would print `false`, rather than `null`
- `ext/filter/tests/filter_var_array_with_ref.phpt` would print `false` in the first line

Reply:
Trying to figure this out now...

Reply:
I've pushed some changes to ext/filter that hopefully make sense. I've also added a warning for the "unknown filter" case, which should make the changes in 057.phpt more obvious.

Any concerns about adding that warning?

Reply:
Thanks for the fixes! This was a more bumpy road for me than what I imagined ^^

At first, I was a bit surprised about the new warnings (I thought that I accidentally removed them previously), but I agree that we should have them :)